api-tests/dev_apis/crypto/test_c024/test_c024.c

```diff
@@ -90,12 +90,18 @@ int32_t psa_aead_encrypt_test(security_t caller)
         TEST_ASSERT_EQUAL(status, PSA_SUCCESS, TEST_CHECKPOINT_NUM(5));
 
         if (is_buffer_empty(check1[i].nonce, check1[i].nonce_length) == TRUE)
+        {
             nonce = NULL;
+            check1[i].nonce_length = 0;
```

Comment:
This is omitted intentionally as it can provide a framework to support testcases passing NULL buffer with a definite size. But the length can be set to zero,  by setting test case with NULL and size as zero

Reply:
I see your point, _however_ AFAIK scenarios such as these where `buffer == NULL && buffer_size > 0` are generally treated as user errors in crypto. Otherwise every function which accepts `buffer == NULL && buffer_size > 0` would have returned an `PSA_ERROR_INVALID_ARGUMENT` error, this is not the case.

Tests 24 & 25 test something else and rely on the fact that these buffers are not accessed before the tested case. On some systems, where there's a separation between SPE & NSPE these buffers _are_ accessed before testing the the key usage, and on TFM systems this leads to reading from memory address `0` which is a security violation.

@gilles-peskine-arm, according to the spec, in general what is the expected behavior for crypto functions which accept a buffer as input where `buffer == NULL && buffer_size > 0`? And specifically for `psa_aead_encrypt` and `psa_aead_decrypt`?